packages/doenetml-worker-rust/lib-doenetml-core/tests/core_basics.rs

```diff
@@ -0,0 +1,53 @@
+mod test_utils;
+use assert_json_diff::assert_json_eq;
+use doenetml_core::{dast::DastRoot, DoenetMLCore};
+use test_utils::*;
+
+#[test]
+fn basic_core_initialization() {
+    let dast_root = dast_root_no_position!("<document>hi there<text>you</text></document>");
+
+    let mut core = DoenetMLCore::new(dast_root, "", "", None);
+
+    let result = core.to_flat_dast();
+
+    let processed_string = to_serde_value!(&result);
+    assert_json_eq!(
+        json!({
+            "type": "root",
+            "children": [0],
+            "elements": [
+                {
+                    "type": "element",
+                    "name": "document",
+                    "attributes": {},
+                    "children": [
+                        "hi there",
+                        1
+                    ],
+                    "data": {
+                        "id": 0,
+                        "action_names": [],
+                        "state": {}
+                    }
+                },
+                {
+                    "type": "element",
+                    "name": "text",
+                    "attributes": {},
+                    "children": [],
+                    "data": {
+                        "id": 1,
+                        "action_names": [],
+                        "state": {
+                            "value": "you",
+                            "text": "you"
```

Comment:
This is a flaw. The `text` state variable should not be marked `for_renderer`.  Should delete the "text" field from "state" and delete line 25 of `text.rs`

Reply:
Found a bug already! I'd say the tests are worthwhile :-)